content/posts/2020-11-11---Poznaj-swoje-IDE.md

```diff
@@ -0,0 +1,72 @@
+- `Alt + Insert` - automatyczne generowanie kodu. W językach obiektowych często gdy tworzymy nową klasę musimy dopisać wiele schematycznych metod jak gettery, settery czy konstruktory. Automatyczne generowanie może łatwo skrócić ten proces.
+- `Ctrl + LPM` - przeniesienie do miejsca deklaracji. Jedna z najbardziej przydatnych komend, o której nie wszycy wiedzą. Jeśli trzymając `Ctrl` naciśniemy LPM na metodę, funckję czy zmienną to zostaniemy przeniesieni do miejsca jej deklaracji w kodzie. 
+- `Ctrl + Shift + F` - znany wszystkim `Ctrl + F` ale dotyczy całego projektu. Wyszukiwanie możemy też ograniczyć do konkternego typu plików czy folderu.
+- `3x Shift` - opcja wyszukiwania podobna do powyższej ale ogranicza się do plików i klas. Dzięki temu często jest wygodniejsza niż `Ctrl + Shift + F`, gdy dokładnie wiemy jakiej klasy szukamy.
+- `Ctrl + R` - pozwala na zastępowanie jednego wyrażenia innym. Bardzo pomocne, gdy zechcemy zmienić nazwę zmiennej na bardziej odpowiednią i potrzebujemy ją zmienić w wielu miejscach.
+
+Chcesz poznać więcej skrótów? Zdecydowaną wiekszość znajdziesz pod tym [linkiem](https://resources.jetbrains.com/storage/products/intellij-idea/docs/IntelliJIDEA_ReferenceCard.pdf?_ga=2.54832427.1862380870.1598387885-556184972.1595441919)
+
+**UWAGA!** Niektóre skróty mogą się kłocić ze skrótami systemowymi. Więcej o tym możesz przeczytać [na stronie JetBrains](https://www.jetbrains.com/help/idea/configuring-keyboard-and-mouse-shortcuts.html#conflicts)
+
+### Zbyt mało opcji? Dodaj Pluginy
+Jakby dostępne możliwości oferowane przez IDE byłyby dla ciebie zbyt małe, zawsze możesz sięgnąć po pluginy, które jeszcze je rozbudowują.
+Wśród używanych i polecanych przeze mnie znjadziemy:
+- **Key Promoter X** - jest to prosty plugin, który pomaga uczyć się skrótów klawiszowych. Za każdym razem jak wybierzesz opcje, na którą jest skrót klawiszowy poinformuje cię o tym. Może być przydatny w przypadku nauki skrótów klawiszowych.
+- **Rainbow Brackets** - ile razy straciłeś czas na szukaniu błędu, który okazał się być źle domknietym nawiasem? Zapewne nie mało.
+Dlatego uważam ten plugin za jeden z najbardziej przydatnych.
+Kolorowanie nawiasów umożliwia na skotrolowanie w ciagu chwili zakresów zakresy pętli czy if-ów w Javie czy C++, a także kolejności operacji w bardziej rozbudowanych warunkach.
+- **SonarLint** - jest jednym z wielu pluginów, których celem jest sprawdzanie czystości kodu. 
+Jeśli znajdzie jakieś problemy z czystością kodu to je zaznaczy byśmy mogli je poprawić, a także poinformuje nas o nich przed wykonaniem commita. 
+Czasem nawet zaznaczy je w taki sposób jakby były bugiem w kodzie, jeśli uzna to za coś poważnego.
+![jetbrains_photo_3](/media/IDE-plugins.png)
+
+### Korzystasz z bazy danych? Podłącz ją bezpośrednio z IDE
+W prawym bocznym panelu jest pasek opcji, na którym znajdziemy `Database`. Po jej kliknięciu pojawi się nam panel boczny. 
```

Comment:
```suggestion
W prawym bocznym panelu jest pasek opcji, na którym znajdziemy `Database`. Po jej kliknięciu pojawi się nam panel boczny. 
```

Z tego zdania wynika, że klikając w panel boczny pojawi nam się panel boczny.

Reply:
Dodać screena, bo to rzeczywiście jest panel boczny który pojawia się po kliknięciu w panel boczny.